components/camel-flink/src/main/java/org/apache/camel/component/flink/FlinkEndpoint.java

```diff
@@ -64,9 +72,12 @@ protected void doStart() throws Exception {
     public Producer createProducer() throws Exception {
         if (endpointType == EndpointType.dataset) {
             return new DataSetFlinkProducer(this);
-        } else {
-            throw new UnsupportedOperationException("datastream not yet supported");
         }
+        else if (endpointType == EndpointType.datastream) {
```

Comment:
Run from command line this goal: mvn install -Psourcecheck in the camel-flink directory
and then fix those checkstyle issues it reports